src/useLocalStorage.ts

```diff
@@ -1,40 +1,54 @@
-import { useEffect, useState } from 'react';
 import { isClient } from './util';
-
-type Dispatch<A> = (value: A) => void;
-type SetStateAction<S> = S | ((prevState: S) => S);
+import { useMemo, useCallback, Dispatch, SetStateAction } from 'react';
+import useEffectOnce from './useEffectOnce';
 
 const useLocalStorage = <T>(key: string, initialValue?: T, raw?: boolean): [T, Dispatch<SetStateAction<T>>] => {
-  if (!isClient) {
+  if (!isClient || !localStorage) {
     return [initialValue as T, () => {}];
   }
+  if (!key && (key as any) !== 0) {
```

Comment:
Can it be a boolean false?

Reply:
Sure, I can protect against that.

Reply:
Not sure what I was asking here looking back.